From a71fc313c4f569be5788caff07ef1fe346842c5b Mon Sep 17 00:00:00 2001
From: "Fujinaka, Todd" <todd.fujinaka@intel.com>
Date: Wed, 23 Oct 2013 05:52:11 +0000
Subject: [PATCH v5 091/164] igb: Don't let ethtool try to write to iNVM in i210/i211

Git-commit: a71fc313c4f569be5788caff07ef1fe346842c5b
Patch-mainline: v3.13-rc1
Reference: fate#317388, bsc#909491
Target: sle11-sp4

Don't let ethtool try to write to iNVM in i210/i211.

This fixes an issue seen by Marek Vasut.

Reported-by: Marek Vasut <marex@denx.de>
Signed-off-by: Todd Fujinaka <todd.fujinaka@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 drivers/net/ethernet/intel/igb/igb_ethtool.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: linux-3.0-SLE11-SP4/drivers/net/ethernet/intel/igb/igb_ethtool.c
===================================================================
--- linux-3.0-SLE11-SP4.orig/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ linux-3.0-SLE11-SP4/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -771,8 +771,10 @@ static int igb_set_eeprom(struct net_dev
 	if (eeprom->len == 0)
 		return -EOPNOTSUPP;
 
-	if (hw->mac.type == e1000_i211)
+	if ((hw->mac.type >= e1000_i210) &&
+	    !igb_get_flash_presence_i210(hw)) {
 		return -EOPNOTSUPP;
+	}
 
 	if (eeprom->magic != (hw->vendor_id | (hw->device_id << 16)))
 		return -EFAULT;
